sstables/sstables.cc

```diff
@@ -2697,7 +2720,7 @@ void sstable::move_to_new_dir_in_thread(sstring new_dir, int64_t new_generation)
 
 std::regex entry_descriptor::la_mc_filename_pattern("(la|mc)-(\\d+)-(\\w+)-(.*)");
 std::regex entry_descriptor::ka_filename_pattern("(\\w+)-(\\w+)-ka-(\\d+)-(.*)");
-std::regex entry_descriptor::sst_dir_pattern(".*/([^/]*)/([^/]+)-[\\da-fA-F]+(?:/staging|/upload|/snapshots/[^/]+)?/?");
+std::regex entry_descriptor::sst_dir_pattern(".*/([^/]*)/([^/]+)-[\\da-fA-F]+(/(staging|upload|snapshots)(/[^/]+)?)?/*");
```

Comment:
I don't see the relaxing, the `staging|upload|snapshot` part is now mandatory. Furthermore there is now an unmatched `(`. Also, this pattern now permits any amount of trailing `/`.